ProcessLib/HeatTransportBHE/BHE/BHE_CXA.h

```diff
@@ -0,0 +1,133 @@
+            GroutParameters const& grout,
+            FlowAndTemperatureControl const& flowAndTemperatureControl,
+            PipeConfigurationCoaxial const& pipes)
+        : BHECommonCoaxial{borehole, refrigerant, grout,
+                           flowAndTemperatureControl, pipes}
+    {
+        // Initialize thermal resistances.
+        auto values = apply_visitor(
+            [&](auto const& control) {
+                return control(refrigerant.reference_temperature,
+                               0. /* initial time */);
+            },
+            flowAndTemperatureControl);
+        updateHeatTransferCoefficients(values.flow_rate);
+    }
+
+    template <int NPoints, typename SingleUnknownMatrixType,
+              typename RMatrixType, typename RPiSMatrixType,
+              typename RSMatrixType>
+    void assembleRMatrices(
+        int const idx_bhe_unknowns,
+        Eigen::MatrixBase<SingleUnknownMatrixType> const& matBHE_loc_R,
+        Eigen::MatrixBase<RMatrixType>& R_matrix,
+        Eigen::MatrixBase<RPiSMatrixType>& R_pi_s_matrix,
+        Eigen::MatrixBase<RSMatrixType>& R_s_matrix) const
+    {
+        switch (idx_bhe_unknowns)
+        {
+            case 0:  // PHI_fig
+                R_matrix.block(0, 2 * NPoints, NPoints, NPoints) +=
+                    -1.0 * matBHE_loc_R;
+                R_matrix.block(2 * NPoints, 0, NPoints, NPoints) +=
+                    -1.0 * matBHE_loc_R;
+
+                R_matrix.block(0, 0, NPoints, NPoints) +=
+                    1.0 * matBHE_loc_R;  // K_i
+                R_matrix.block(2 * NPoints,
+                               2 * NPoints,
+                               NPoints,
+                               NPoints) += 1.0 * matBHE_loc_R;  // K_ig
+                return;
+            case 1:  // PHI_ff
+                R_matrix.block(0, NPoints, NPoints, NPoints) +=
+                    -1.0 * matBHE_loc_R;
+                R_matrix.block(NPoints, 0, NPoints, NPoints) +=
+                    -1.0 * matBHE_loc_R;
+
+                R_matrix.block(0, 0, NPoints,
+                               NPoints) += 1.0 * matBHE_loc_R;  // K_i1
+                R_matrix.block(NPoints, NPoints, NPoints, NPoints) +=
+                    1.0 * matBHE_loc_R;  // K_o
+                return;
+            case 2:  // PHI_gs
+                R_s_matrix += matBHE_loc_R;
+
+                R_pi_s_matrix.block(2 * NPoints, 0, NPoints, NPoints) +=
+                    -1.0 * matBHE_loc_R;
+
+                R_matrix.block(2 * NPoints, 2 * NPoints, NPoints,
+                               NPoints) += matBHE_loc_R;  // K_ig
+                return;
+            default:
+                OGS_FATAL(
+                    "Error!!! In the function BHE_CXA::assembleRMatrices, "
+                    "the index of bhe unknowns is out of range! ");
+        }
+    }
+
+public:
+    std::array<double, number_of_unknowns> cross_section_areas_coaxial()
```

Comment:
function names must be in camelBack case.

Reply:
changed.